| <img src="/assets/dipi.png" alt="DIPI Badge" width="150">      | Data Integrity and Pipeline Integration (DIPI) Working Group |
| -------------- | -------------------- |
| **Meeting**    | 12 Show and Text     |
| **Date**       | 2021-05-11           |
| **Attendance** | SN RC                |
| **Away**       |                      |

## Proposed agenda

* Development roundup
* Pipeline updates
* Upcoming threats
* AOB

## Previous actions

* Nothing to discuss

## Special items

### Phylopipe2

* RC reports Phylopipe2.0 has been running nearly daily and is being published to a date stamped beta directory (which isnâ€™t being linked to latest yet)
    * RC has improved performance by tweaking thread count, skipping the USHER "soft update" and only forcibly adds sequences from the past 30 days into the tree
    * Output formats should be compatible with Microreact now
    * A smooth tree update completes in under 2.5 hours
    * Fixes for USHER related segfault (based on command line options) and tree annotation (spurious ' causing issues in NEXUS)
* RC reports next task is to add a step to rescale the tree after using USHER to change branch lengths into floats <1 (USHER co-erces lengths to integers proportional to mutation count)

### Updates to adm1

* SN still to implement new `adm1` in Majora and friends, but waiting for confirmation from TRC on the status of approvals

### ENA consensus submissions

* SN reports slow turn around times for consensus submissions to make it through to INSDC.
* Test submissions have been going reasonably well, intermittent issues with the ENA API cause uploads to periodically fail but resubmission seems to work fine.
* SN needs to enable an opt-in on Majora to allow sites to use automated ENA consensus submissions but has not yet done this as some documentation for sites to fill in the `bioinfo_pipe_*` fields correctly is needed
    * SN was hoping someone from the DWG would take an interest in writing this

## Standing items

### Current development work

* SN has mostly been working on a side project to rapidly find edit distance between multiple sequence alignment sequences this week.
    * Initially used `edlib` and `xxhash` but found that `edlib` was being too clever to handle the MSA (was trying to find indels when the alignment was already done)
        * `edlib` and `xxhash` enabled a sequence to be searched against the COG-UK data set in a few minutes (and one sequence could be handled by one process, so this scaled linearly)
    * To improve speed, SN is working on `warphog` (https://github.com/SamStudio8/warphog) to calculate Hamming distance of fixed length sequences in a CUDA kernel
        * Initial testing indicates warphog can handle around 4 million sequence pairs a second on a hilariously overpowered Nvidia GV100
        * 1000 sequences can be checked against the COG-UK data set in around 2 minutes (a significant improvement on the CPU bound `edlib` work)
    * SN hoped to use this to precalculate a distance matrix for the COG-UK dataset, before realising that 500,000 squared is a very large number
        * Most likely use-case is to search a selection of sequences against the COG-UK dataset, so the CUDA version is likely overkill -- will take a look at also improving the CPU-based version in case that is useful
* RC has also been working on `scorpio` to extract lineage/group defining mutations (primarily with `pangolin` users in mind)
    * scorpio is a general tool/non covid specific to define/classify/haplotype constellations of mutations
        * Define takes a column of nucleotide mutations, and a column with group ids and collects the common/intermediate frequency mutations and translates them into aa-space definitions
        * Haplotype and classify take ref-coordinate MSA as input
        * Haplotype creates a string for each constellation with the alt call (or - for ref) so can see at a glance what mutations are present/not
        * Classify uses some rules to determine if sequence should be classified as a constellation
    * Some overlaps with PHE variant definitions, but scorpio is not just for VOCs/VUIs
        * SN happy to help if there is anything we can do to help make the PHE variant definitions more portable for tools like scorpio
* RC will release `pangolin` with an `USHER` mode (likely this week)
    * Pangolin has been struggling with the number of convergent mutations cropping up which define lineages. This is being tackled in multiple ways:
        * new decision tree/lineage regression model in development for pangoLEARN
        * new mode using usher to assign lineages based off a tree as an alternative
        * replacing the post-hoc tests done for problematic VOCs within pangolin with scorpio    

### Pipeline updates

* GISAID pipeline fell over on May 10th as the API could not be reached, this was transient and was restarted without trouble (https://github.com/COG-UK/dipi-group/issues/70)
* Datapipe received some updates because we were getting sequences drop out during alignment steps due to sequence divergence not meeting threshold in minimap2 preset - move from asm5 to asm20


### Threats

* SN and RC have made some space available by deleting nextflow caches

### AOB

* No AOB

## Action summary

| Issue | For | Action          | Timescale          |
|-------|-----|-----------------|--------------------|
|||||
